qa/rspec/matchers/cli_matchers.rb

```diff
@@ -0,0 +1,31 @@
+# encoding: utf-8
+RSpec::Matchers.define :be_successful do
+  match do |actual|
+    actual.exit_status == 0
+  end
+end
+
+RSpec::Matchers.define :fail_and_output do |expected_output|
+  match do |actual|
+    actual.exit_status == 1 && actual.stderr =~ expected_output
+  end
+end
+
+RSpec::Matchers.define :run_successfully_and_output do |expected_output|
+  match do |actual|
+    (actual.exit_status == 0 || actual.exit_status.nil?) && actual.stdout =~ expected_output
+  end
+end
+
+RSpec::Matchers.define :have_installed? do |name,*args|
+  match do |actual|
+    version = args.first
```

Comment:
you can actually have a cleaner interface to this if you use something like http://www.relishapp.com/rspec/rspec-expectations/v/3-4/docs/custom-matchers/define-matcher-with-fluent-interface

this will allow you have something like

```ruby
  expect(logstash).to have_installed?("logstash-input-foo").with_version("0.1.1")
```

what do you think?